### PR TITLE
Feature FDA receiver access request of change

### PR DESCRIPTION
**Pull request by AngieCristina**
**Implementation Notes**

- Added Calendar to show the schedule change request

### CODE REVIEW

File: App/src/components/RequestModal.js [hunk 1]
```diff
@@ -1,8 +1,169 @@
...
+            id: reqFind.requested_id,
+            date: reqFind.date_Shift,
+            shift_id: reqFind.requested_Shift_id
+          }
+        ]
+      : [];
+    setShiftsClicked(data);
+  }, [request, id]);
+
+  const forecasts = [
```
Comment (MariethPerez) [inline App/src/components/RequestModal.js]: I think that this data 'forecasts' shouldn't be in Request Modal.

File: App/src/components/RequestModal.js [hunk 2]
```diff
@@ -1,8 +1,169 @@
...
+  const workShiftConcat = events.reduce((groups, event) => {
+    return {
+      ...groups,
+      [event.user_id]: groups[event.user_id]
+        ? groups[event.user_id].concat(event.workShifts)
+        : event.workShifts
+    };
+  }, {});
+
+  const forecastsConcat = forecasts.reduce((groups, forecast) => {
```
Comment (MariethPerez) [inline App/src/components/RequestModal.js]: I think that this function 'forecastsConcat' shouldn't be in RequestModal.

File: App/src/components/RequestModal.js [hunk 3]
```diff
@@ -44,22 +209,35 @@ function RequestModal({ request, isOpen, onRequestClose, id, setRequests }) {
...
+          ) : (
+            <div key={e.id}>{""}</div>
+          )
         )}
       </div>
+      <Calendar
+        workShiftConcat={workShiftConcat}
+        frontdesks={frontdesks}
+        calcDay={calcDay}
+        start={0}
```
Comment (MariethPerez) [inline App/src/components/RequestModal.js:135]: I think that the 'start' and 'end' variables should indicate the week the shift change belongs to.

File: App/src/views/RequestsView.js [hunk 4]
```diff
@@ -14,28 +14,33 @@ function RequestsView() {
...
-    const id = event.target.dataset.id
+
+  function handleRequestSchedule(id) {
     setId(id);
     setModalOpen(true);
   }
+
   if (!user) return <Redirect to="login" noThrow />;
+
+  if (frontdesks.length === 0) return "Cargando...";
```
Comment (MariethPerez) [inline App/src/views/RequestsView.js:41]: It is good!!!

**Review by DiegoCuevas — APPROVED** [review]

**Review by MariethPerez — APPROVED** [review]

**Pull request merged**